addons/mail/static/src/core/web/messaging_menu.js

```diff
@@ -37,14 +38,16 @@ export class MessagingMenu extends Component {
         });
     }
 
-    async beforeOpen() {
-        if (
-            !this.store.discuss.inbox.isLoaded &&
-            this.store.discuss.inbox.status !== "loading" &&
-            this.store.discuss.inbox.counter !== this.store.discuss.inbox.messages.length
-        ) {
-            this.threadService.fetchNewMessages(this.store.discuss.inbox);
-        }
+    beforeOpen() {
+        this.messagingService.isReady.then(() => {
```

Comment:
Note: the code was delayed through `await` in override before calling super, now it is made non-blocking.

The goal is to remove the dependency on `isReady` in future commits.